src/codecs/tga/decoder.rs

```diff
@@ -247,15 +248,14 @@ impl<R: Read + Seek> TgaDecoder<R> {
             // of 0 would be pointless.
             if (run_packet & 0x80) != 0 {
                 // high bit set, so we will repeat the data
-                let repeat_count = ((run_packet & !0x80) + 1) as usize;
-                let mut data = Vec::with_capacity(self.bytes_per_pixel);
+                let repeat_count = ((run_packet & !0x80) + 1) as usize * self.bytes_per_pixel;
                 self.r
                     .by_ref()
                     .take(self.bytes_per_pixel as u64)
-                    .read_to_end(&mut data)?;
-                for _ in 0usize..repeat_count {
-                    pixel_data.extend_from_slice(&data);
-                }
+                    .read_to_end(&mut pixel)?;
+                let data = pixel.iter().cycle().take(repeat_count);
```

Comment:
This is just slightly obscure, I would appreciate a comment. And maybe move the multiplication with `self.bytes_per_pixel` into the argument here because `repeat_count` is a term that comes from tga itself and this way we preserve its original meaning and value.